app/src/main/java/com/android/sample/model/video/VideoViewModel.kt

```diff
@@ -0,0 +1,58 @@
+// portions of this code were done with the help of ChatGPT and GitHub Copilot
+
+package com.android.sample.model.video
+
+import android.net.Uri
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.withContext
+
+class VideoViewModel(private val videoRepository: VideoRepository) : ViewModel() {
+
+  private val _videoUrls = MutableStateFlow<List<String>>(emptyList())
+  val videoUrls: StateFlow<List<String>>
+    get() = _videoUrls.asStateFlow()
+
+  private val _uploadSuccess = MutableStateFlow<String?>(null)
+  val uploadSuccess: StateFlow<String?>
+    get() = _uploadSuccess.asStateFlow()
+
+  private val _error = MutableStateFlow<String?>(null)
+  val error: StateFlow<String?>
+    get() = _error.asStateFlow()
+
+  // Calls repository to upload video and updates the state
+  fun uploadVideo(videoUri: Uri) {
+    videoRepository.uploadVideo(
+        videoUri,
+        { downloadUrl -> _uploadSuccess.value = downloadUrl },
+        { exception -> _error.value = "Upload failed: ${exception.message}" })
+  }
+
+  // Calls repository to fetch all video URLs and updates the state
+  suspend fun loadVideos() {
+    withContext(Dispatchers.IO) {
+      videoRepository.getVideoUrls(
+          { urls -> _videoUrls.update { urls } },
+          { exception -> _error.update { "Failed to load videos: ${exception.message}" } })
+    }
+  }
+
```

Comment:
Maybe add in the KDoc an example of how to actually use VideoViewModel, or a little bit more exaplanation of what it does for the one calling it. When and how should it be used?